external-import/cve/src/connector/cve_connector.py

```diff
@@ -0,0 +1,286 @@
+
+            if current_state is not None and "last_run" in current_state:
+                last_run = current_state["last_run"]
+
+                msg = "[CONNECTOR] Connector last run: " + datetime.utcfromtimestamp(
+                    last_run
+                ).strftime("%Y-%m-%d %H:%M:%S")
+                self.helper.log_info(msg)
+            else:
+                last_run = None
+                msg = "[CONNECTOR] Connector has never run..."
+                self.helper.log_info(msg)
+
+            """
+            ======================================================
+            Main process if connector successfully works
+            ======================================================
+            """
+            # Initiate work_id to track the job
+            work_id = self._initiate_work(current_time)
+
+            """
+            ================================================================
+            If the connector never runs, import the most recent CVEs
+            from the last max_date_range (can be configured) to now
+            ================================================================
+            """
+            if last_run is None:
+                self._import_recent(now, work_id)
+
+                """
+                =================================================================
+                If the connector never runs and user wants to pull CVE history
+                =================================================================
+                """
+                if last_run is None and self.config.pull_history:
+                    start_date = datetime(self.config.history_start_year, 1, 1)
+                    end_date = now
+                    self._import_history(start_date, end_date, work_id)
+
+                self.update_connector_state(current_time, work_id)
+
+                """
+                ===================================================================
+                Import CVEs from the last run to now if maintain data is True
+                If the connector runs, and last run is more than current interval
+                ===================================================================
+                """
+            elif (
+                last_run is not None
+                and self.config.maintain_data
+                and (current_time - last_run) >= int(self.config.interval)
+            ):
+                self._maintain_data(now, last_run, work_id)
+                self.update_connector_state(current_time, work_id)
+
+            else:
+                new_interval = self.config.interval - (current_time - last_run)
+                new_interval_in_hours = round(new_interval / 60 / 60, 2)
+                self.helper.log_info(
+                    "[CONNECTOR] Connector will not run, next run in: "
+                    + str(new_interval_in_hours)
+                    + " hours"
+                )
+
+            time.sleep(5)
```

Comment:
Why 5 seconds when the interval is defined by the user?

Reply:
The interval defined by the user is the interval to check and import new CVEs, this time sleep is only to properly stop the connector on keyboard interruption or system exit :)